File: eco-core/core-plugin/src/main/java/com/willfp/ecoarmor/commands/CommandEcoarmor.java
package com.willfp.ecoarmor.commands;


import com.willfp.eco.core.EcoPlugin;
import com.willfp.eco.core.command.CommandHandler;
import com.willfp.eco.core.command.impl.PluginCommand;
import org.jetbrains.annotations.NotNull;

public class CommandEcoarmor extends PluginCommand {
    /**
     * Instantiate a new command handler.
     *
     * @param plugin The plugin for the commands to listen for.
     */
    public CommandEcoarmor(@NotNull final EcoPlugin plugin) {
        super(plugin, "ecoarmor", "ecoarmor.command.ecoarmor", false);

        this.addSubcommand(new CommandReload(plugin))
                .addSubcommand(new CommandGive(plugin));
    }

    @Override
    public CommandHandler getHandler() {
        return (sender, args) -> {
            sender.sendMessage(this.getPlugin().getLangYml().getMessage("invalid-command"));
        };
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/ecoarmor/commands/CommandEcoarmor.kt
package com.willfp.ecoarmor.commands

import com.willfp.eco.core.command.impl.PluginCommand
import com.willfp.ecoarmor.sets.ArmorSets
import com.willfp.libreforge.LibReforgePlugin
import com.willfp.libreforge.lrcdb.CommandExport
import com.willfp.libreforge.lrcdb.CommandImport
import com.willfp.libreforge.lrcdb.ExportableConfig
import org.bukkit.command.CommandSender

class CommandEcoarmor(plugin: LibReforgePlugin) : PluginCommand(plugin, "ecoarmor", "ecoarmor.command.ecoarmor", false) {
    init {
        addSubcommand(CommandReload(plugin))
            .addSubcommand(CommandGive(plugin))
            .addSubcommand(CommandImport("sets", plugin))
            .addSubcommand(CommandExport(plugin) {
                ArmorSets.values().map {
                    ExportableConfig(
                        it.id,
                        it.config
                    )
                }
            })
    }

    override fun onExecute(sender: CommandSender, args: List<String>) {
        sender.sendMessage(
            plugin.langYml.getMessage("invalid-command")
        )
    }
}
